### PR TITLE
Environment (core, builtin_env, builtin_export, builtin_unset)

### PR DESCRIPTION
**Pull request by taaae**

### CODE REVIEW

File: src/environment/environment_utils.c [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	size_t	environ_size;
+	char	**environ;
+	char	**new_environ;
+
+	environ = get_environ();
+	environ_size = 0;
+	while (*environ++)
+		environ_size++;
+	new_environ = ft_calloc(environ_size + 2, sizeof(char *));
+	environ -= (environ_size + 1);
```
Comment (poroshinleonid) [inline src/environment/environment_utils.c:45]: UB when environ_size is 0.

**Review by poroshinleonid — CHANGES_REQUESTED** [review]

File: src/environment/environment_utils.c (hunk 1 above)
Comment (taaae) [reply src/environment/environment_utils.c:45]: I think`while (*environ++)` will increment pointer by 1 is size is 0, might be wrong

File: src/environment/environment.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+{
+	size_t	environ_size;
+	size_t	index;
+	char	**environ;
+
+	environ_size = 0;
+	while (*envp++)
+		environ_size++;
+	environ = ft_calloc(environ_size + 1, sizeof(char *));
+	envp -= (environ_size + 1);
```
[the remark this answers is not in the document]
Comment (taaae) [reply src/environment/environment.c:30]: same here

File: src/environment/environment.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+void	extend_environ(void);
+
+void	ft_initenv(char **envp)
+{
+	size_t	environ_size;
+	size_t	index;
+	char	**environ;
+
+	environ_size = 0;
+	while (*envp++)
```
[the remark this answers is not in the document]
Comment (taaae) [reply src/environment/environment.c:27]: yeah, it's just from `int main(int argc, char **argv, char **envp)`

**Pull request merged**